### PR TITLE
Portal Storm Coherency Pass

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Balance "Portal Storm Coherency Pass"

#### Purpose of change
Portal Storms are a bit all over the place and didn't have a good coherent balance level, numerous people were approaching them from very different perspectives and producing various outcomes. This is confusing and frustrating from a player perspective. 

#### Describe the solution
First off, added a new design doc that covers Portal Storm balance. This doc discusses most all the ideas presented in this PR, instead of regurgitating that I'm gonna just discuss the changes. [Formatted doc](https://github.com/CleverRaven/Cataclysm-DDA/blob/8d05345ad46437f6d5559bb6b3ee0639631968ac/doc/PORTAL_STORM_BALANCE_AND_DESIGN.md)

Added new Ire EOC `EOC_PORTAL_EFFECTS_IRE` which causes you to build up ire when outside
Added snippets for this EOC

Migrated all remaining portal effects to either Passive or Active, each of these has their own EOC `EOC_PORTAL_EFFECTS_ACTIVE` and `EOC_PORTAL_EFFECTS_PASSIVE`

Passive Effects:

EOC_PORTAL_GRASS_CHANGE: No change
EOC_PORTAL_LIGHT: No change
EOC_PORTAL_ARTIFACT_WEAK: No change
EOC_PORTAL_ARTIFACT_MEDIUM: consolidated with weak
EOC_PORTAL_ARTIFACT_STRONG: consolidated with weak
EOC_PORTAL_ABSENCE: No longer moves
EOC_PORTAL_IMPOSSIBLE_SHAPE: No longer moves, massive range on attack, builds to madness doesn't immediately make you go mad
EOC_PORTAL_SHIFTING_MASS: No changes (least happy with this one, I think it should be worsened and made active)
EOC_PORTAL_MEMORIES: No sight, no ears
EOC_PORTAL_GIANT_APPENDAGE: No sight, no ears

Active Effects:
EOC_PORTAL_TELEPORT_STUCK_START: No change
EOC_PORTAL_PAIN: No change
EOC_PORTAL_HALLUCINATOR: Actually opens doors now
EOC_PORTAL_SMOKE: No change
EOC_PORTAL_SWARM_STRUCTURE: Deals way less fatigue and pain, does more damage to itself when it teleports or spawns blades
EOC_PORTAL_NPC: No change

#### Testing
Tested each entity individually and being in portal storms a bit.

#### Additional context
image

### CODE REVIEW

**Comment by GuardianDll** [comment]
What IRE stands for?

**Comment by bombasticSlacks** [comment]
image

**Comment by GuardianDll** [comment]
Oh, i thought its some abbreviation

File: data/json/monster_special_attacks/spells.json [hunk 1]
```diff
@@ -588,6 +601,17 @@
         "decay_start": "24 hours"
       },
       { "math": [ "u_pain()", "+=", "3" ] }
+      { "math": [ "u_madness", "=", "0" ] },
+      {
+        "u_message": "Your mind lurches, the shapes and figures dance across your vision, you feel like your head is going to explode."
+      }
+    ],
+    "false_effect": [
+      { "math": [ "u_pain()", "+=", "3" ] }
```
Comment (bombasticSlacks) [inline data/json/monster_special_attacks/spells.json:610]:
```suggestion
      { "math": [ "u_pain()", "+=", "3" ] },
```

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -588,6 +601,17 @@
         "decay_start": "24 hours"
       },
       { "math": [ "u_pain()", "+=", "3" ] }
```
Comment (bombasticSlacks) [inline data/json/monster_special_attacks/spells.json:603]:
```suggestion
      { "math": [ "u_pain()", "+=", "3" ] },
```

**Comment by SariusSkelrets** [comment]
Looks good!

However, the chunks should be in the active effects since they both inflict physical injuries, are always agressive and are described as reacting to the player's presence to the point of not appearing if the player isn't outside

That would have a side effect of making them rarer but it would also make them easier to engage as the ire cost would impede the creation of other active effects. Their strength, tankiness, reward on kill and the strongest version's electrical field warrant it

**Comment by bombasticSlacks** [comment]
> However, the chunks should be in the active effects since they both inflict physical injuries, are always agressive and are described as reacting to the player's presence to the point of not appearing if the player isn't outside

ahhh, when I tested them in game I thought they were blind, but it was just nighttime + clairvoyance.

File: doc/PORTAL_STORM_BALANCE_AND_DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+* Portal storms shouldn't be forced rest: It should be *possible* for a strong character to maneuver in a portal storm. It can be risky, but shouldn't be impossible.
+* Portal storm encounters don't need to be rewarding: Some encounters will be, but not every encounter needs a carrot. The variance of portal storms means that traveling the storms is a gamble, sometimes you get lucky, other times you do not.
+* Portal storms should only be *actively* dangerous with creatures hunting you when you go outside (more on this in balance).
+* Portal storms are trying to destroy humans, and warp the world: Portal storms should also really only be dangerous to humans, creatures that can percieve the otherworldly nature. Animals and the dead should have little to no interest in the other side and vice versa.
+* Don't be annoying: Nothing ruins ambience as fast as tedium. Don't spam effects, don't waste the players time. Portal Storms already *show* the player they are dangerous, we do not need to *tell* the player they are scary or weird.
```
Comment (Maddremor) [inline doc/PORTAL_STORM_BALANCE_AND_DESIGN.md:14]: "Tedium" is pretty vague here and generally just used for heckling. Consider "repetition" instead.

File: data/json/monsters/nether.json [hunk 4]
```diff
@@ -1378,6 +1378,7 @@
     "death_function": { "corpse_type": "NO_CORPSE", "message": "it just disappeared." },
     "flags": [
       "NO_BREATHE",
+      "CAN_OPEN_DOORS",
```
Comment (GuardianDll) [inline data/json/monsters/nether.json:1381]: image

File: doc/PORTAL_STORM_BALANCE_AND_DESIGN.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# Portal Storm Design
+Portal storms are otherworldly colisions. Evil and malicious entities from inexplainable spaces are pouring into our world. However this doesn't mean it's a good way to imediately kill the character.
```
Comment (actually-a-cat) [inline doc/PORTAL_STORM_BALANCE_AND_DESIGN.md:2]:
```suggestion
Portal storms are otherworldly collisions. Evil and malicious entities from inexplainable spaces are pouring into our world. However this doesn't mean it's a good way to immediately kill the character.
```

File: doc/PORTAL_STORM_BALANCE_AND_DESIGN.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+## Reminders
+Some things to keep in mind when working on Portal Storms:
+* Portal storms are weather: You are not writing boss fights, or end game encounters, you are writing and implementing weather effects. Try not to lose track of that.
+* Portal storms shouldn't be forced rest: It should be *possible* for a strong character to maneuver in a portal storm. It can be risky, but shouldn't be impossible.
+* Portal storm encounters don't need to be rewarding: Some encounters will be, but not every encounter needs a carrot. The variance of portal storms means that traveling the storms is a gamble, sometimes you get lucky, other times you do not.
+* Portal storms should only be *actively* dangerous with creatures hunting you when you go outside (more on this in balance).
+* Portal storms are trying to destroy humans, and warp the world: Portal storms should also really only be dangerous to humans, creatures that can percieve the otherworldly nature. Animals and the dead should have little to no interest in the other side and vice versa.
```
Comment (actually-a-cat) [inline doc/PORTAL_STORM_BALANCE_AND_DESIGN.md:13]:
```suggestion
* Portal storms are trying to destroy humans, and warp the world: Portal storms should also really only be dangerous to humans, creatures that can perceive the otherworldly nature. Animals and the dead should have little to no interest in the other side and vice versa.
```

File: doc/PORTAL_STORM_BALANCE_AND_DESIGN.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+* Themed storms. Currently we have one eldritch themed storm with a mix of everything, having a number of themed storms with different effects and thematic voices would be great.
+
+# Portal Storm Balance
+There are a few seperate concepts dictating portal storms.
+
+## Ire
+Being outside in a portal storm should draw the attention of the malicious enities being pulled into our realm by the storm. This is refered to in the code and this document as `ire`.
+
+## Passive vs Active
+Portal storms should represent two seperate ideas.
```
Comment (actually-a-cat) [inline doc/PORTAL_STORM_BALANCE_AND_DESIGN.md:31]:
```suggestion
Portal storms should represent two separate ideas.
```

File: doc/PORTAL_STORM_BALANCE_AND_DESIGN.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+A few of the additional goals for storms are as follows:
+
+* Localized and trackable storms, similar to a game like Death Stranding where you can detect and avoid storm areas, or even drive out of them.
+* Themed storms. Currently we have one eldritch themed storm with a mix of everything, having a number of themed storms with different effects and thematic voices would be great.
+
+# Portal Storm Balance
+There are a few seperate concepts dictating portal storms.
+
+## Ire
+Being outside in a portal storm should draw the attention of the malicious enities being pulled into our realm by the storm. This is refered to in the code and this document as `ire`.
```
Comment (actually-a-cat) [inline doc/PORTAL_STORM_BALANCE_AND_DESIGN.md:28]:
```suggestion
Being outside in a portal storm should draw the attention of the malicious entities being pulled into our realm by the storm. This is referred to in the code and this document as `ire`.
```

**Pull request merged**